packages/security/src/DefaultCredentialManager.ts

```diff
@@ -49,156 +51,224 @@ import * as keytar from "keytar"; // Used for typing purposes only
-                additionalDetails: this.getMissingEntryMessage(account)
-            });
-        }
+  /**
+   * Reference to the lazily loaded keytar module.
+   */
+  private keytar: typeof keytar;
+
+  /**
+   * Errors that occurred while loading keytar will be stored in here.
+   *
+   * Every method of this class should call the {@link checkForKeytar} method before proceeding. It
+   * is this method that will check for keytar and throw this error if it was detected that keytar
+   * wasn't loaded.
+   *
+   * @private
+   */
+  private loadError: ImperativeError;
+
+  /**
+   * Pass-through to the superclass constructor.
+   *
+   * @param {string} service The service string to send to the superclass constructor.
+   * @param {string} displayName The display name for this credential manager to send to the superclass constructor
+   */
+  constructor(service: string, displayName: string = "Zowe Built-in Credential Manager") {
+    // Always ensure that a manager instantiates the super class, even if the
+    // constructor doesn't do anything. Who knows what things might happen in
+    // the abstract class initialization in the future.
+    super(service, displayName);
+  }
+
+  /**
+   * Called by {@link CredentialManagerFactory.initialize} before the freeze of the object. This
+   * gives us a chance to load keytar into the class before we are locked down. If a load failure
+   * occurs, we will store the error and throw it once a method of this class tries to execute. This
+   * prevents a missing keytar module from stopping all operation of the cli.
+   *
+   * In the future, we could go even further to have keytar load into a sub-object of this class so
+   * that the load doesn't hold up the main class execution.
+   *
+   * @returns {Promise<void>} A promise that the function has completed.
+   */
+  public async initialize(): Promise<void> {
+      try {
+        // tslint:disable-next-line:no-implicit-dependencies
+        this.keytar = await import("keytar");
+      } catch (error) {
+          this.loadError = new ImperativeError({
+              msg: "Keytar not Installed",
+              causeErrors: error
+          });
+      }
+  }
+
+  /**
+   * Calls the keytar deletePassword service with {@link DefaultCredentialManager#service} and the
+   * account passed to the function by Imperative.
+   *
+   * @param {string} account The account for which to delete the password
+   *
+   * @returns {Promise<void>} A promise that the function has completed.
+   *
+   * @throws {@link ImperativeError} if keytar is not defined.
+   * @throws {@link ImperativeError} when keytar.deletePassword returns false.
+   */
+  protected async deleteCredentials(account: string): Promise<void> {
+    this.checkForKeytar();
+
+    if (!await this.deleteCredentialsHelper(account)) {
+      throw new ImperativeError({
+        msg: "Unable to delete credentials.",
+        additionalDetails: this.getMissingEntryMessage(account)
+      });
```

Comment:
In the SCS plugin we made a change to stop throwing this error. Reasons for doing so are listed in [this PR comment](https://github.com/zowe/zowe-cli-scs-plugin/pull/7).

Not necessarily suggesting any action here, as perhaps the scenarios listed there are no longer applicable now? If we're storing only a single secure value, I don't think we'll be deleting values from the secure vault too frequently anymore.

Reply:
Same comment as below.